docs/user/troubleshooting_job.md

```diff
@@ -17,51 +17,197 @@
+
+If it retries many times, and isn't caused by limited resource, administrators of OpenPAI may needs to check what happens.
+
+![retry](imgs/web_job_detail_retry.png)
+
+### Job runs slowly
+
+The running speed of job is subjective sometime, so it needs more fact and data to estimate what's expected speed, before trying to fix it. Besides to fix code to improve speed, below is several reasons that may make job slowly in OpenPAI.
+
+1. GPU is not enabled. Some framework, like TensorFlow, needs to install GPU package to enable GPU. In most case, the log of framework shows if GPU or CPU is enabled. Some framework, like PyTorch, code needs to be updated to GPU, and there is API to check if GPU is used. Learn more from [how to check job log](#how-to-check-job-log).
+
+2. Resource bottleneck. Computing resource is not the only potential bottleneck, IO and memory capacity are also bottleneck sometime. When job is running in OpenPAI, metrics can be used to analyze bottleneck. Refer to [how to check job metrics](#how-to-check-job-metrics) for more information.
+
+### Job is failed
+
+Job failure can be caused by many reasons.
+
+1. **System failures like over maximum resources**. If a job requests more than maximum resources, which the OpenPAI cluster can provide, the job fails shortly. For example, if the cluster has only 32 cores of CPU, but user requests 48 cores in a job configuration, it causes job failure.
+
+   For this kind of system failures, there is no resource assigned, no IP, ports and GPUs as below.
+
+   ![over requested 1](imgs/web_job_details_over1.png)
+
+   Click *View Application Summary*, there is exception with message like below. it specifies which resource is over than maximum number. In below case, it requests 48 virtual cores, but 24 virtual cores are maximum available number in this cluster.
+
+   ![over requested 1](imgs/web_job_details_over2.png)
+
+   Note, there may be other kinds of system failures happening. If no resource assigned, click *View Application Summary* to get details of system failure.
+
+2. **User level failures**. If IP address and GPU are assigned, it's user level failure in most case. In this case, log provides details of failure. Learn more from [how to check job log](#how-to-check-job-log).
+
+   Note, OpenPAI detects job failure by returns non-zero exit code of task instance.
+
+   ![job link](imgs/web_job_details_exitcode.png)
+
+## Guidance
+
+### How to view job metrics
+
+- Click *Go to Job Metrics Page* as below in job details page, if some tasks are assigned IP, and ports.
+
+![job link](imgs/web_job_details_metrics.png)
+
+- A new page is opened and show metrics of this job.
+
+![job link](imgs/web_job_metrics.png)
+
+- The *memory usage*, *network traffic* and *disk bandwidth* uses absolute value. It's easy to understand.
```

Comment:
Should notice: PAI sacrifices network metric accuracy by using host network in docker container for better performance, so, PAI can not relay on docker virtual bridge to get accurate network traffic consumption.

PAI uses another way: get all connections and traffic used by each connection, then break down into each container. This logic is implemented in [job-exporter](https://github.com/Microsoft/pai/blob/master/src/job-exporter/src/network.py). So, if your connection is short-lived or not happened in container namespace(for example, NFS mount), traffic in this connection will not captured by job-exporter and hence will show 0 in panel.

Reply:
Thanks, I will update it accordingly.